google/cloud/spanner/internal/build_info.cc.in

```diff
@@ -24,10 +24,6 @@ namespace spanner {
 inline namespace SPANNER_CLIENT_NS {
 namespace internal {
 
-std::string CompilerId() { return R"""(@CMAKE_CXX_COMPILER_ID@)"""; }
-
-std::string CompilerVersion() { return R"""(@CMAKE_CXX_COMPILER_VERSION@)"""; }
-
```

Comment:
I really like this idea! I think this is a good direction.

A couple thoughts, but since I can't comment on unedited lines (in GH's review tool), I'll put all the comments here:

1. Let's move all the compile-time detected functions into the new file. Which means, let's also move `CompilerFeatures()` and `LangaugeVersion()`.

2. Let's rename `compiler_version` -> `compiler_info`, to make it a little more generic (so LanguageVersion makes sense in it).

3. Also create `compiler_info.h` to declare the functions. The result here will be that `compiler_info` and `build_info*` will be independent. 

4. Move the parts of the unit test that for the "compiler_info" things into "compiler_info_test.cc"

How does this sound?

Reply:
Done.